pkgs/development/compilers/ghc/common-hadrian.nix

```
@@ -639,7 +639,7 @@ stdenv.mkDerivation (
       "--with-ffi-includes=${targetLibs.libffi.dev}/include"
       "--with-ffi-libraries=${targetLibs.libffi.out}/lib"
     ]
-    ++ lib.optionals (targetPlatform == hostPlatform && !enableNativeBignum) [
+    ++ lib.optionals (!enableNativeBignum && !targetPlatform.isGhcjs && !targetPlatform.isWasm) [
       "--with-gmp-includes=${targetLibs.gmp.dev}/include"
       "--with-gmp-libraries=${targetLibs.gmp.out}/lib"
```

Doesn't that imply gmp for static? Would that be a problem because of license?

---

I'll defer to @sternenseemann on this one. I am happy with the remainder of the PR, I think, but am really unsure about the last commit.

---

- This doesn't change whether GMP is used or not (this depends solely on `enableNativeBignum`). The point of this condition was probably not to use the `--with-gmp-*` flags if we're building a cross compiler since this can cause to incorrect paths or `-L` flags being used by Hadrian since it doesn't actually distinguish between host and target gmp. Instead the idea is to just let the cc wrapper figure it out. The downside of this is usually that the package db entries don't have the necessary paths. Would need some blaming to figure out why this was added originally.
- @alexfmpe We did you end up enabling the flag again? Does this solve a specific problem?
- The `targetPlatform.isGhcjs` is redundant since we already flick on `enableNativeBignum` for that. We could do the same for wasm, but I don't entirely see the point since it's purely theoretical.

```suggestion
    ++ lib.optionals (!enableNativeBignum) [
      "--with-gmp-includes=${targetLibs.gmp.dev}/include"
      "--with-gmp-libraries=${targetLibs.gmp.out}/lib"
```